### ISSUE TITLE
Problem after update to 0.9.8

### ISSUE THREAD

**Issue by S474N** [DaveMDS/node-red-contrib-tasmota#26]
Node-RED version: v1.2.9 , docker version
Node.js  version: v12.20.1
Linux 5.10.17-v7l+ arm LE

After update plugin node-red-contrib-tasmota to newest version 0.9.8 have problem with rebooting/cycling Nodered with this log:

```
2021-04-06T08:40:34.808416873Z 6 Apr 10:40:34 - [info] [mqtt-broker:c31e24e2.791568] Connected to broker: mqtt://10.0.0.1:1883
2021-04-06T08:40:34.890535879Z 6 Apr 10:40:34 - [red] Uncaught Exception:
2021-04-06T08:40:34.891029779Z 6 Apr 10:40:34 - TypeError: ts.startsWith is not a function
2021-04-06T08:40:34.891092463Z     at matchTopic (/usr/src/node-red/node_modules/@node-red/nodes/core/network/10-mqtt.js:38:20)
2021-04-06T08:40:34.891120259Z     at MqttClient.handler (/usr/src/node-red/node_modules/@node-red/nodes/core/network/10-mqtt.js:319:25)
2021-04-06T08:40:34.891145259Z     at MqttClient.emit (events.js:326:22)
2021-04-06T08:40:34.891170018Z     at MqttClient._handlePublish (/usr/src/node-red/node_modules/mqtt/lib/client.js:1277:12)
2021-04-06T08:40:34.891193981Z     at MqttClient._handlePacket (/usr/src/node-red/node_modules/mqtt/lib/client.js:410:12)
2021-04-06T08:40:34.891217462Z     at work (/usr/src/node-red/node_modules/mqtt/lib/client.js:321:12)
2021-04-06T08:40:34.891377256Z     at processTicksAndRejections (internal/process/task_queues.js:79:11)
2021-04-06T08:40:34.936478783Z npm ERR! code ELIFECYCLE
2021-04-06T08:40:34.937217068Z npm ERR! errno 1
2021-04-06T08:40:34.950677039Z npm ERR! node-red-docker@1.2.9 start: `node $NODE_OPTIONS node_modules/node-red/red.js $FLOWS "--userDir" "/data"`
2021-04-06T08:40:34.950860462Z npm ERR! Exit status 1
2021-04-06T08:40:34.951248956Z npm ERR! 
2021-04-06T08:40:34.952061074Z npm ERR! Failed at the node-red-docker@1.2.9 start script.
2021-04-06T08:40:34.952132925Z npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
2021-04-06T08:40:34.969871981Z 
2021-04-06T08:40:34.970310086Z npm ERR! A complete log of this run can be found in:
2021-04-06T08:40:34.970806004Z npm ERR!     /data/.npm/_logs/2021-04-06T08_40_34_953Z-debug.log
2021-04-06T08:40:36.174071677Z 
2021-04-06T08:40:36.174150250Z > node-red-docker@1.2.9 start /usr/src/node-red
2021-04-06T08:40:36.174178879Z > node $NODE_OPTIONS node_modules/node-red/red.js $FLOWS "--userDir" "/data"
2021-04-06T08:40:36.174205712Z 
```

**Comment by S474N** [comment]
Tested recovery volumes/nodered folder from backup and update only tasmota plugin, and have this error:
![image](https://user-images.githubusercontent.com/16512337/113689492-b6d67400-96ca-11eb-9721-a583d2aae19f.png)

```
-----------------------------------------------------------
2021-04-06T09:24:08.082Z Install : node-red-contrib-tasmota 0.9.8

2021-04-06T09:24:06.842Z npm install --no-audit --no-update-notifier --no-fund --save --save-prefix=~ --production node-red-contrib-tasmota@0.9.8
2021-04-06T09:24:19.855Z [err] npm
2021-04-06T09:24:19.856Z [err]  
2021-04-06T09:24:19.856Z [err] ERR!
2021-04-06T09:24:19.856Z [err]  
2021-04-06T09:24:19.856Z [err] code
2021-04-06T09:24:19.858Z [err]  EEXIST
2021-04-06T09:24:19.859Z [err] npm
2021-04-06T09:24:19.859Z [err]  
2021-04-06T09:24:19.860Z [err] ERR!
2021-04-06T09:24:19.860Z [err]  
2021-04-06T09:24:19.860Z [err] path
2021-04-06T09:24:19.860Z [err]  /data/node_modules/.bin/mqtt_pub
2021-04-06T09:24:19.894Z [err] npm
2021-04-06T09:24:19.894Z [err]  
2021-04-06T09:24:19.894Z [err] ERR!
2021-04-06T09:24:19.895Z [err]  Refusing to delete /data/node_modules/.bin/mqtt_pub: is outside /data/node_modules/mqtt and not a link
2021-04-06T09:24:19.895Z [err] npm
2021-04-06T09:24:19.895Z [err]  
2021-04-06T09:24:19.895Z [err] ERR!
2021-04-06T09:24:19.895Z [err]  File exists: /data/node_modules/.bin/mqtt_pub
2021-04-06T09:24:19.896Z [err] npm
2021-04-06T09:24:19.896Z [err]  
2021-04-06T09:24:19.896Z [err] ERR!
2021-04-06T09:24:19.896Z [err]  Remove the existing file and try again, or run npm
2021-04-06T09:24:19.896Z [err] npm
2021-04-06T09:24:19.896Z [err]  
2021-04-06T09:24:19.897Z [err] ERR!
2021-04-06T09:24:19.897Z [err]  with --force to overwrite files recklessly.
2021-04-06T09:24:19.913Z [err] 
2021-04-06T09:24:19.914Z [err] npm ERR! A complete log of this run can be found in:
2021-04-06T09:24:19.914Z [err] npm ERR!
2021-04-06T09:24:19.914Z [err]      /data/.npm/_logs/2021-04-06T09_24_19_904Z-debug.log
2021-04-06T09:24:19.935Z rc=1
```

**Comment by Vitek-1977** [comment]
Hello, same problem here, node red crashed after configuring the tasmota-mqtt broker. Had to delete the plugin manually, restart nodered, remove all tasmota nodes plus the configuration node and install other tasmota node. Same config as S474N above.

**Comment by DaveMDS** [comment]
I have no idea what is going on here, I updated all my instances without issues.

After upgrading you need to reconfigure the broker node, as we do not use anymore the NodeRed broker config.
From the CHANGELOG:
!!! WARNING !!! With this release you need to setup again the mqtt-broker node for all your deployed nodes. As we do not use anymore the mqtt-broker-node included in NodeRed.

**Comment by Vitek-1977** [comment]
Obviously, the mqtt broker conflicts somehow with mosquitto...


![image](https://user-images.githubusercontent.com/82035731/113711301-1477ba00-96e5-11eb-8c0f-2f86f562ecc6.png)

**Comment by DaveMDS** [comment]
So strange I did not get this errors in my instance...
I just released a fast fix in the new 0.9.9
Now I'm checking the broker configuration before doing anything

please let me know if this work better for you

Sorry for the inconvenience, I'm trying to break stuff now to release soon a 1.0 stable release

**Comment by S474N** [comment]
So tested:
/data/node_modules/.bin/mqtt_pub
/data/node_modules/.bin/mqtt_sub
/data/node_modules/.bin/mqtt

![image](https://user-images.githubusercontent.com/16512337/113716393-d8dfee80-96ea-11eb-8026-7c00257bee8f.png)

If I manually delete these files, then will update of plugin continue. Then restarted and configured new MQTT broker. 

BUT!! If I have two same nodes, than first is ok and second (with same configuration) have "`Wrong config`".

![image](https://user-images.githubusercontent.com/16512337/113716615-12185e80-96eb-11eb-8ea0-04b3315261a5.png)

**Comment by DaveMDS** [comment]
hi, thanks for testing this

The first issue (existing files) are outside the scope of the TasmotaNodes, I don't know what those files are, 
they are not part of my distribution package... maybe they come from the mqtt library that is required? no idea here

For the second one: 
I think you did not update the second node configuration. Double-click the node, press Done and redeploy.
This must be done for ALL tasmota nodes already deployed (also if the config SEEMS already correct, it must be saved)

**Comment by hamster65** [comment]
I updated to 0.9.9, created a new mqtt broker and went into all mqtt nodes to select the new broker. Some nodes show "wrong config", others are online and working, others are offline, no matter what I do. I played with the settings and created a second broker. Somehow, one of the offline nodes was working. Then I switched back to the first broker and it was offline again. Switched back to second.... still offline. Odd....

**Comment by hamster65** [comment]
The switch node shows "offline", but when I send commands to it, the switch... switches. There is no output from the node.

**Comment by DaveMDS** [comment]
@hamster65 The "wrong config" in nodes means you did not saved correctly the node config, the check is super simple and cannot go wrong, double check the config for those nodes
The offline status of nodes instead is strange, it smell to me like a duplicated ClientID, make sure all your mqtt client use a different ClientID.

**Comment by hamster65** [comment]
Right, the "wrong config" is gone now after saving the node ocnfig again. It still shows offline. It's odd, some of those nodes were "online" the day before. I also entered a random ID for the 2 brokers and rebootet the PI, still offline.

**Comment by hamster65** [comment]
I defined 2 brokers and several nodes (new ones and old ones which I created before the update). When I select a different broker on one node, other nodes go online or offline. I thought it happens when an "old" node is using that broker but that is not it.... it is more random. I have 5 nodes now with 2 different brokers, all going to the same tasmota plug. Some are online, others offline. The only consistend thing is: Nodes created before the update are always offline.

**Comment by hamster65** [comment]
I got it running by deleting all tasmota brokers, all tasmota nodes and then adding the nodes again to the flow. Copying the nodes did not work, it worked only with new nodes. As long as one old node was present, even the new nodes did not work. I also noticed that when deploying, node red took a very long time to finish as long as the old nodes were present. The log showd that it was waiting for mqtt to disconnect. Once all nodes were replaced, it worked fast again.
